Code/Engine/Core/Scripting/ScriptComponent.h

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <Core/Scripting/ScriptClassResource.h>
+#include <Core/World/EventMessageHandlerComponent.h>
+#include <Foundation/Types/RangeView.h>
+
+using ezScriptComponentManager = ezComponentManager<class ezScriptComponent, ezBlockStorageType::FreeList>;
+
+class EZ_CORE_DLL ezScriptComponent : public ezEventMessageHandlerComponent
+{
+  EZ_DECLARE_COMPONENT_TYPE(ezScriptComponent, ezEventMessageHandlerComponent, ezScriptComponentManager);
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezComponent
+
+protected:
+  virtual void SerializeComponent(ezWorldWriter& stream) const override;
+  virtual void DeserializeComponent(ezWorldReader& stream) override;
+  virtual void Initialize() override;
+  virtual void Deinitialize() override;
+  virtual void OnActivated() override;
+  virtual void OnDeactivated() override;
+  virtual void OnSimulationStarted() override;
+
+  /*virtual bool OnUnhandledMessage(ezMessage& msg, bool bWasPostedMsg) override;
+  virtual bool OnUnhandledMessage(ezMessage& msg, bool bWasPostedMsg) const override;
+
+  bool HandleUnhandledMessage(ezMessage& msg, bool bWasPostedMsg);*/
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezEventMessageHandlerComponent
+
+protected:
+  // virtual bool HandlesEventMessage(const ezEventMessage& msg) const override;
+
+  //////////////////////////////////////////////////////////////////////////
+  // ezScriptComponent
+public:
+  ezScriptComponent();
+  ~ezScriptComponent();
+
+  void BroadcastEventMsg(ezEventMessage& msg);
```

Comment:
Why is this here? Is this a message handler?

Reply:
Some as above, no message handling or sending yet, might be needed in the future.